op-node/rollup/driver/state.go

```diff
@@ -457,6 +458,28 @@ func (s *Driver) eventLoop() {
 	}
 }
 
+func (s *Driver) syncStep(ctx context.Context) error {
+	// If we don't need to call FCU to restore unsafeHead using backupUnsafe, keep going b/c
+	// this was a no-op(except correcting invalid state when backupUnsafe is empty but TryBackupUnsafeReorg called).
+	if fcuCalled, err := s.engineController.TryBackupUnsafeReorg(ctx); fcuCalled {
+		// If we needed to perform a network call, then we should yield even if we did not encounter an error.
+		return err
+	}
+	// If we don't need to call FCU, keep going b/c this was a no-op. If we needed to
+	// perform a network call, then we should yield even if we did not encounter an error.
+	if err := s.engineController.TryUpdateEngine(ctx); !errors.Is(err, derive.ErrNoFCUNeeded) {
+		return err
+	}
+	// Trying unsafe payload should be done before safe attributes
+	// It allows the unsafe head to move forward while the long-range consolidation is in progress.
+	if err := s.clSync.Proceed(ctx); err != io.EOF {
+		// EOF error means we can't process the next unsafe payload. Then we should process next safe attributes.
+		return err
+	}
+	s.metrics.SetDerivationIdle(false)
```

Comment:
Its a bit error prone to not have it be obvious where this is set to true

Reply:
In https://github.com/ethereum-optimism/optimism/pull/10643 I am changing it a little, to encapsulate it better to the derivation pipeline, but still not perfect. For now we have to keep it to keep the metric around.